internal/venice-common/src/main/java/com/linkedin/venice/kafka/KafkaClientFactory.java

```diff
@@ -66,63 +62,53 @@ public PubSubConsumer getConsumer(Properties props, KafkaPubSubMessageDeserializ
         kafkaPubSubMessageDeserializer);
   }
 
-  public Consumer<byte[], byte[]> getRawBytesKafkaConsumer() {
-    Properties props = getConsumerProps();
-    // This is a temporary fix for the issue described here
-    // https://stackoverflow.com/questions/37363119/kafka-producer-org-apache-kafka-common-serialization-stringserializer-could-no
-    // In our case "org.apache.kafka.common.serialization.ByteArrayDeserializer" class can not be found
-    // because class loader has no venice-common in class path. This can be only reproduced on JDK11
-    // Trying to avoid class loading via Kafka's ConfigDef class
-    props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class);
-    props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class);
-    // Increase receive buffer to 1MB to check whether it can solve the metadata timing out issue
-    props.put(ConsumerConfig.RECEIVE_BUFFER_CONFIG, 1024 * 1024);
-
-    return getKafkaConsumer(props);
-  }
-
-  @Deprecated
-  public Consumer<KafkaKey, KafkaMessageEnvelope> getRecordKafkaConsumer() {
-    Properties props = getConsumerProps();
-    props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, KafkaKeySerializer.class);
-    props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, OptimizedKafkaValueSerializer.class);
-    if (kafkaMessageEnvelopeSchemaReader.isPresent()) {
-      props.put(VENICE_SCHEMA_READER_CONFIG, kafkaMessageEnvelopeSchemaReader.get());
-    }
-
-    return getKafkaConsumer(props);
-  }
-
   private <K, V> Consumer<K, V> getKafkaConsumer(Properties properties) {
     Properties propertiesWithSSL = setupSSL(properties);
     return new KafkaConsumer<>(propertiesWithSSL);
   }
 
-  public KafkaAdminWrapper getWriteOnlyKafkaAdmin(Optional<MetricsRepository> optionalMetricsRepository) {
-    return createAdminClient(getWriteOnlyAdminClass(), optionalMetricsRepository, "WriteOnlyKafkaAdminStats");
+  public KafkaAdminWrapper getWriteOnlyKafkaAdmin(
+      Optional<MetricsRepository> optionalMetricsRepository,
+      PubSubTopicRepository pubSubTopicRepository) {
+    return createAdminClient(
+        getWriteOnlyAdminClass(),
+        optionalMetricsRepository,
+        "WriteOnlyKafkaAdminStats",
+        pubSubTopicRepository);
   }
 
-  public KafkaAdminWrapper getReadOnlyKafkaAdmin(Optional<MetricsRepository> optionalMetricsRepository) {
-    return createAdminClient(getReadOnlyAdminClass(), optionalMetricsRepository, "ReadOnlyKafkaAdminStats");
+  public KafkaAdminWrapper getReadOnlyKafkaAdmin(
```

Comment:
I don't think we really need different variations(read-only and write-only) of Admin API. I know they have been there in the code already, but IIUC they are not necessary anymore. Can we just keep once of them?